### DISCUSSION TITLE
Multiple data series

### DISCUSSION THREAD

**Discussion by johnohems** [pymeasure/pymeasure#605] [Q&A]
Problem:
I am unable to run a single procedure that plots multiple, unconnected series (maybe even different colours) in the results graph while saving data to one file.

Context:
I'm attempting to plot spectrums across a range of temperatures. Ideally this would all be done in one procedure to save all data in one file, however I cannot find a way to separate the data series during graphing of results within one procedure. I can't find a solution in the docs but may have missed it, is this possible?

**Comment by msmttchr** [comment]
Currently pymeasure procedure define a vector of values (vector can have an arbitrary number of elements and each element has a name) and in the procedure loop it is expected that a vector is generated which is in turn saved to a file (as a new row) and plotted in the graph.
A limitation is that in the graph you can plot only two elements of the vector (one being X and another being Y) according to user selection (see also #393).
On the CSV file saved, you will have all the vector elements.
In summary, plotting more than one curve at time is not possible. Saving more than one curve in the CSV is possible.
Hope this helps, but I am not sure if it answers your question properly.
